### PR TITLE
feat: add stats option to analyze command for performance insights

### PR DESCRIPTION
**Pull request by matthisholleville**
Closes # 

## 📑 Description

Introduced a new feature to the analyze command that enables users to print detailed performance statistics of each analyzer. This enhancement aids in debugging and understanding the time taken by various components during analysis, providing valuable insights for performance optimization.

## ✅ Checks

- [x] My pull request adheres to the code style of this project
- [x] My code requires changes to the documentation
- [x] I have updated the documentation as required
- [x] All the tests have passed

## ℹ Additional Information

```bash
(matthisholleville) ➜  k8sgpt git:(main) ✗ time ./bin/k8sgpt analyze -s
The stats mode allows for debugging and understanding the time taken by an analysis by displaying the statistics of each analyzer.
- Analyzer Ingress took 47.125583ms 
- Analyzer PersistentVolumeClaim took 53.009167ms 
- Analyzer CronJob took 57.517792ms 
- Analyzer Deployment took 156.6205ms 
- Analyzer Node took 160.109833ms 
- Analyzer ReplicaSet took 245.938333ms 
- Analyzer StatefulSet took 448.0455ms 
- Analyzer Pod took 5.662594708s 
- Analyzer Service took 38.583359166s 
```

### CODE REVIEW

**Comment by matthisholleville** [comment]
Linked to this issue https://github.com/k8sgpt-ai/k8sgpt/pull/1237

**Comment by AlexsJones** [comment]
If we do `-o json -s` what will happen, will it add a field into the json object?

**Comment by matthisholleville** [comment]
When this option is selected, the results are not displayed. Only the statistics are shown. If we try to display both, in the case of an analysis with many results, it becomes very difficult to visualize the statistics.

**Comment by AlexsJones** [comment]
> The current implementation does not support json as an output.

Might be worth considering if we want to show the results and time taken

**Comment by matthisholleville** [comment]
@AlexsJones stats implementation is global to analyzer. The results display the errors reported by an analyzer. In my opinion, these are two different notions. We could also display the stats in json format, if that's useful.

File: README.md [hunk 1]
```diff
@@ -318,6 +318,22 @@ _Analysis with custom headers_
 k8sgpt analyze --explain --custom-headers CustomHeaderKey:CustomHeaderValue
 ```
 
+_Print analysis stats_
+
+```
+k8sgpt analyze -s
```
Comment (AlexsJones) [inline README.md:324]: I can't see a world where you'd want to run this without capturing the output in some format. 
But for those that do, perhaps... `k8sgpt analyze --stats --no-output` ?

Comment (matthisholleville) [reply README.md:325]: This simply makes the result of the analysis unreadable with the stats.

File: pkg/analysis/analysis.go [hunk 2]
```diff
@@ -296,24 +273,41 @@ func (a *Analysis) RunAnalysis() {
...
+	stat := common.AnalysisStats{
+		Analyzer:     filter,
+		DurationTime: elapsedTime,
+	}
+
+	mutex.Lock()
+	defer mutex.Unlock()
+
+	if err != nil {
+		a.Stats = append(a.Stats, stat)
```
[the remark this answers is not in the document]
Comment (AlexsJones) [reply pkg/analysis/analysis.go:302]: @matthisholleville really excited to get this in - can you respond to this?

**Review by AlexsJones — APPROVED** [review]
Overall this is a really great PR and an enormous step in the right direction. Thanks for this @matthisholleville

**Review by arbreezy — APPROVED** [review]

**Pull request merged**